geowebcache/rest/src/main/java/org/geowebcache/rest/service/SeedService.java

```diff
@@ -94,27 +91,52 @@ public ResponseEntity<?> getRunningLayerTasks(HttpServletRequest request, String
         try {
             XStream xs = new GeoWebCacheXStream(new JsonHierarchicalStreamDriver());
             long[][] list;
-            if (null == layer) {
-                list = seeder.getStatusList();
-            } else {
-                try {
-                    seeder.findTileLayer(layer);
-                } catch (GeoWebCacheException e) {
-                    HttpHeaders headers = new HttpHeaders();
-                    headers.setContentType(MediaType.TEXT_PLAIN);
-                    return new ResponseEntity<>(e.getMessage(), headers, HttpStatus.BAD_REQUEST);
-                }
-                list = seeder.getStatusList(layer);
+            try {
+                list = findTileLayer(layer);
+            } catch (GeoWebCacheException e) {
+                return getBadRequestResponse(
+                        MediaType.TEXT_PLAIN, e.getMessage(), HttpStatus.BAD_REQUEST);
             }
             JSONObject obj = new JSONObject(xs.toXML(list));
-            HttpHeaders headers = new HttpHeaders();
-            headers.setContentType(MediaType.APPLICATION_JSON);
-            return new ResponseEntity<>(obj.toString(), headers, HttpStatus.OK);
+            return getBadRequestResponse(MediaType.APPLICATION_JSON, obj.toString(), HttpStatus.OK);
```

Comment:
I don't know if it was intentional or a mistake, but it seems strange to get a valid result (JSONObject obj) and then pass it through a getBadRequestResponse method. I guess the method is not misnamed, it should be getStringResponse or something similar.

Reply:
Sorry. Yes, it was a mistake because I was fixing this in parralel with another task and failed to minimize context switch, thus lost some attention. Fixed name.